### PR TITLE
feat(ui): Pod View in UI to improve usability of Large Applications

### PR DESCRIPTION
**Pull request by rbreeze**
pod-view

Checklist:

* [x] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [x] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [x] Optional. My organization is added to USERS.md.
* [ ] I've signed the CLA and my build is green ([troubleshooting builds](https://argoproj.github.io/argo-cd/developer-guide/ci/)).

### CODE REVIEW

**Comment by rbreeze** [comment]
pod-view-live

**Comment by alexec** [comment]
this looks ace

**Comment by simster7** [comment]
This looks great!

**Comment by rbreeze** [comment]
Additional iconography coming to improve usability for color blind users.

Screen Shot 2020-11-04 at 1 28 19 PM

Simulated view with deuteranopia:
Screen Shot 2020-11-04 at 1 28 13 PM

**Comment by rbreeze** [comment]
Updated view with node information and improved iconography: 

Screen Shot 2020-11-12 at 11 08 30 AM

**Comment by rbreeze** [comment]
Preview of ability to sort pods by parent resource

Screen Shot 2020-11-13 at 1 19 09 PM

**Comment by jsoref** [comment]
Could you at least check:
- [ ] Does this PR require documentation updates?

**Comment by jsoref** [comment]
Fwiw, it's probably better to move the images from the discussion into the description (speaking from experience landing a major feature to jenkins).

File: ui/src/app/applications/components/application-pod-view/pod-view.tsx [hunk 1]
```diff
@@ -0,0 +1,390 @@
...
+                                load={async () => {
+                                    return podPrefs.sortMode === 'node' ? services.applications.getNodes(this.props.app.metadata.name) : null;
+                                }}>
+                                {nodes => {
+                                    return (
+                                        <div className='nodes-container'>
+                                            {this.processTree(podPrefs.sortMode, nodes).map(group => (
+                                                <div className='node white-box' key={group.name}>
+                                                    <div className='node__container--header' onClick={() => this.props.onItemClick(group.fullName)} style={{cursor: 'pointer'}}>
+                                                        <div style={{display: 'flex', alignItems: 'center'}}>
```
Comment (alexmt) [inline ui/src/app/applications/components/application-pod-view/pod-view.tsx:90]: Please let's make sure node containers have the same width:


image

**Review by alexmt — CHANGES_REQUESTED** [review]
Awesome work! Added few mostly minor comments. Also please run `yarn lint --fix --project .` in `ui` folder to fix linter errors.

File: ui/src/app/applications/components/application-pod-view/pod-view.scss [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+        align-items: center;
+        display: flex;
+        margin-bottom: 1em;
+        &__section {
+            margin: 0 10px;
+        }
+    }
+}
+
+.nodes-container {
```
Comment (alexmt) [inline ui/src/app/applications/components/application-pod-view/pod-view.scss:19]: Please make `nodes-container` and `node` sub-classes of `pod-view` . E.g. `pod-view__nodes-container` and `pod-view__node`

**Review by alexmt — CHANGES_REQUESTED** [review]

File: ui/src/app/applications/components/application-resource-tree/application-resource-tree.tsx [hunk 3]
```diff
@@ -274,11 +274,13 @@ function renderResourceNode(props: ApplicationResourceTreeProps, id: string, nod
                         {node.createdAt || props.app.metadata.creationTimestamp}
                     </Moment>
                 ) : null}
-                {(node.info || []).filter(tag => !tag.name.includes('Resource.')).map((tag, i) => (
-                    <span className='application-resource-tree__node-label' title={`${tag.name}:${tag.value}`} key={i}>
-                        {tag.value}
-                    </span>
-                ))}
+                {(node.info || [])
```
Comment (alexmt) [inline ui/src/app/applications/components/application-resource-tree/application-resource-tree.tsx:277]: Pods now have a lot of "info items": 

image

I suggest to show top 4 and add tooltip with all info items.

**Review by alexmt — CHANGES_REQUESTED** [review]

**Pull request merged**